<!DOCTYPE html>
<html>
<head>
    <link rel="stylesheet" type="text/css" href="../style.css">
</head>
<body>


<h1>CDSSequence.java</h1>

<table class="src">



<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_1'/>
1
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_1'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>/*</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_2'/>
2
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_2'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *                    BioJava development code</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_3'/>
3
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_3'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_4'/>
4
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_4'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * This code may be freely distributed and modified under the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_5'/>
5
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_5'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * terms of the GNU Lesser General Public Licence.  This should</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_6'/>
6
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_6'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * be distributed with the code.  If you do not have a copy,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_7'/>
7
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_7'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * see:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_8'/>
8
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_8'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_9'/>
9
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_9'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.gnu.org/copyleft/lesser.html</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_10'/>
10
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_10'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_11'/>
11
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_11'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * Copyright for this code is held jointly by the individual</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_12'/>
12
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_12'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * authors.  These should be listed in @author doc comments.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_13'/>
13
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_13'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_14'/>
14
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_14'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * For more information on the BioJava project and its aims,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_15'/>
15
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_15'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * or to join the biojava-l mailing list, visit the home page</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_16'/>
16
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_16'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * at:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_17'/>
17
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_17'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_18'/>
18
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_18'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.biojava.org/</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_19'/>
19
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_19'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_20'/>
20
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_20'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * Created on DATE</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_21'/>
21
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_21'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_22'/>
22
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_22'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_23'/>
23
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_23'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>package org.biojava.nbio.core.sequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_24'/>
24
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_24'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_25'/>
25
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_25'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_26'/>
26
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_26'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.DNACompoundSet;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_27'/>
27
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_27'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_28'/>
28
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_28'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.template.CompoundSet;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_29'/>
29
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_29'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_30'/>
30
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_30'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_31'/>
31
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_31'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * Represents a exon or coding sequence in a gene. It has a parent {@link TranscriptSequence}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_32'/>
32
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_32'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * where a TranscriptSequence is the child of a GeneSequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_33'/>
33
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_33'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * Not important for protein construction but the phase is used if outputting the gene</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_34'/>
34
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_34'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * to a gff3 file. {@link http://www.sequenceontology.org/gff3.shtml}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_35'/>
35
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_35'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * @author Scooter Willis</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_36'/>
36
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_36'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_37'/>
37
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_37'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>public class CDSSequence extends DNASequence {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_38'/>
38
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_38'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_39'/>
39
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_39'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;//private static final Logger log = Logger.getLogger(CDSSequence.class.getName());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_40'/>
40
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_40'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;Integer phase = 0; // 0, 1, 2</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_41'/>
41
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_41'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;TranscriptSequence parentTranscriptSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_42'/>
42
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_42'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_43'/>
43
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_43'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_44'/>
44
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_44'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_45'/>
45
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_45'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param parentSequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_46'/>
46
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_46'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param bioBegin</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_47'/>
47
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_47'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param bioEnd</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_48'/>
48
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_48'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param phase</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_49'/>
49
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_49'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_50'/>
50
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_50'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;public CDSSequence(TranscriptSequence parentSequence, int bioBegin, int bioEnd, int phase) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_51'/>
51
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_51'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;parentTranscriptSequence = parentSequence;</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_52'/>
52
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_52'>1</a>
<span>
1. <init> : removed call to org/biojava/nbio/core/sequence/CDSSequence::setParentSequence &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;this.setParentSequence(parentTranscriptSequence);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_53'/>
53
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_53'>1</a>
<span>
1. <init> : removed call to org/biojava/nbio/core/sequence/CDSSequence::setBioBegin &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;setBioBegin(bioBegin);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_54'/>
54
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_54'>1</a>
<span>
1. <init> : removed call to org/biojava/nbio/core/sequence/CDSSequence::setBioEnd &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;setBioEnd(bioEnd);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_55'/>
55
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_55'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;this.phase = phase;</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_56'/>
56
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_56'>1</a>
<span>
1. <init> : removed call to org/biojava/nbio/core/sequence/CDSSequence::setCompoundSet &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;this.setCompoundSet(DNACompoundSet.getDNACompoundSet());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_57'/>
57
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_57'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_58'/>
58
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_58'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_59'/>
59
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_59'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_60'/>
60
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_60'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;@Override</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_61'/>
61
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_61'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public int getLength() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_62'/>
62
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_62'>3</a>
<span>
1. getLength : Replaced integer subtraction with addition &rarr; NO_COVERAGE<br/>
2. getLength : Replaced integer addition with subtraction &rarr; NO_COVERAGE<br/>
3. getLength : replaced return of integer sized value with (x == 0 ? 1 : 0) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return Math.abs(this.getBioEnd() - this.getBioBegin()) + 1;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_63'/>
63
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_63'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_64'/>
64
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_64'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_65'/>
65
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_65'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_66'/>
66
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_66'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_67'/>
67
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_67'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return get the phase</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_68'/>
68
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_68'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_69'/>
69
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_69'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public Integer getPhase() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_70'/>
70
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_70'>1</a>
<span>
1. getPhase : mutated return of Object value for org/biojava/nbio/core/sequence/CDSSequence::getPhase to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return phase;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_71'/>
71
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_71'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_72'/>
72
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_72'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_73'/>
73
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_73'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_74'/>
74
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_74'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_75'/>
75
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_75'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return get the strand</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_76'/>
76
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_76'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_77'/>
77
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_77'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public Strand getStrand() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_78'/>
78
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_78'>1</a>
<span>
1. getStrand : mutated return of Object value for org/biojava/nbio/core/sequence/CDSSequence::getStrand to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return parentTranscriptSequence.getStrand();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_79'/>
79
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_79'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_80'/>
80
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_80'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_81'/>
81
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_81'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_82'/>
82
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_82'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * A CDS sequence if negative stranded needs to be reverse complement</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_83'/>
83
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_83'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * to represent the actual coding sequence. When getting a ProteinSequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_84'/>
84
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_84'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * from a TranscriptSequence this method is callled for each CDSSequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_85'/>
85
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_85'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * {@link http://www.sequenceontology.org/gff3.shtml}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_86'/>
86
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_86'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * {@link http://biowiki.org/~yam/bioe131/GFF.ppt}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_87'/>
87
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_87'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return coding sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_88'/>
88
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_88'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_89'/>
89
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_89'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public String getCodingSequence() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_90'/>
90
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_90'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;String sequence = this.getSequenceAsString(getBioBegin(), getBioEnd(), getStrand());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_91'/>
91
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_91'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_92'/>
92
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_92'>1</a>
<span>
1. getCodingSequence : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;if (getStrand() == Strand.NEGATIVE) {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_93'/>
93
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_93'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;//need to take complement of sequence because it is negative and we are returning a coding sequence</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_94'/>
94
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_94'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;StringBuilder b = new StringBuilder(getLength());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_95'/>
95
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_95'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;CompoundSet&#60;NucleotideCompound&#62; compoundSet = this.getCompoundSet();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_96'/>
96
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_96'>2</a>
<span>
1. getCodingSequence : changed conditional boundary &rarr; NO_COVERAGE<br/>
2. getCodingSequence : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;for (int i = 0; i &#60; sequence.length(); i++) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_97'/>
97
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_97'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;String nucleotide = String.valueOf(sequence.charAt(i));</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_98'/>
98
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_98'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;NucleotideCompound nucleotideCompound = compoundSet.getCompoundForString(nucleotide);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_99'/>
99
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_99'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;b.append(nucleotideCompound.getComplement().getShortName());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_100'/>
100
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_100'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_101'/>
101
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_101'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;sequence = b.toString();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_102'/>
102
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_102'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_103'/>
103
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_103'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;//  sequence = sequence.substring(phase);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_104'/>
104
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_104'>1</a>
<span>
1. getCodingSequence : mutated return of Object value for org/biojava/nbio/core/sequence/CDSSequence::getCodingSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return sequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_105'/>
105
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_105'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@6b34f713_106'/>
106
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_106'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>}</span></pre></td></tr>



<tr><td></td><td></td><td><h2>Mutations</h2></td></tr>

<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6b34f713_52'>52</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_52'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b><init><br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/CDSSequence::setParentSequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6b34f713_53'>53</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_53'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b><init><br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/CDSSequence::setBioBegin &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6b34f713_54'>54</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_54'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b><init><br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/CDSSequence::setBioEnd &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6b34f713_56'>56</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_56'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b><init><br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/CDSSequence::setCompoundSet &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6b34f713_62'>62</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_62'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getLength<br/><b>Killed by : </b>none</span></span> Replaced integer subtraction with addition &rarr; NO_COVERAGE</p> <p class='NO_COVERAGE'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>getLength<br/><b>Killed by : </b>none</span></span> Replaced integer addition with subtraction &rarr; NO_COVERAGE</p> <p class='NO_COVERAGE'><span class='pop'>3.<span><b>3</b><br/><b>Location : </b>getLength<br/><b>Killed by : </b>none</span></span> replaced return of integer sized value with (x == 0 ? 1 : 0) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6b34f713_70'>70</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_70'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getPhase<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/CDSSequence::getPhase to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6b34f713_78'>78</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_78'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getStrand<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/CDSSequence::getStrand to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6b34f713_92'>92</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_92'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getCodingSequence<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6b34f713_96'>96</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_96'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getCodingSequence<br/><b>Killed by : </b>none</span></span> changed conditional boundary &rarr; NO_COVERAGE</p> <p class='NO_COVERAGE'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>getCodingSequence<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@6b34f713_104'>104</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@6b34f713_104'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getCodingSequence<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/CDSSequence::getCodingSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>

</table>


<h2>Active mutators</h2>
<ul>
<li class='mutator'>INCREMENTS_MUTATOR</li>
<li class='mutator'>VOID_METHOD_CALL_MUTATOR</li>
<li class='mutator'>RETURN_VALS_MUTATOR</li>
<li class='mutator'>MATH_MUTATOR</li>
<li class='mutator'>NEGATE_CONDITIONALS_MUTATOR</li>
<li class='mutator'>INVERT_NEGS_MUTATOR</li>
<li class='mutator'>CONDITIONALS_BOUNDARY_MUTATOR</li>

</ul>

<h2>Tests examined</h2>
<ul>
</ul>

<br/>

Report generated by <a href='http://pitest.org'>PIT</a> 1.4.0

</body>
</html>