<!DOCTYPE html>
<html>
<head>
    <link rel="stylesheet" type="text/css" href="../style.css">
</head>
<body>


<h1>TranscriptSequence.java</h1>

<table class="src">



<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_1'/>
1
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_1'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>/*</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_2'/>
2
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_2'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *                    BioJava development code</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_3'/>
3
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_3'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_4'/>
4
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_4'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * This code may be freely distributed and modified under the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_5'/>
5
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_5'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * terms of the GNU Lesser General Public Licence.  This should</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_6'/>
6
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_6'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * be distributed with the code.  If you do not have a copy,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_7'/>
7
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_7'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * see:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_8'/>
8
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_8'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_9'/>
9
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_9'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.gnu.org/copyleft/lesser.html</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_10'/>
10
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_10'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_11'/>
11
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_11'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * Copyright for this code is held jointly by the individual</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_12'/>
12
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_12'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * authors.  These should be listed in @author doc comments.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_13'/>
13
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_13'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_14'/>
14
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_14'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * For more information on the BioJava project and its aims,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_15'/>
15
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_15'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * or to join the biojava-l mailing list, visit the home page</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_16'/>
16
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_16'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * at:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_17'/>
17
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_17'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_18'/>
18
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_18'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.biojava.org/</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_19'/>
19
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_19'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_20'/>
20
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_20'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * Created on DATE</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_21'/>
21
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_21'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_22'/>
22
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_22'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_23'/>
23
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_23'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>package org.biojava.nbio.core.sequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_24'/>
24
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_24'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_25'/>
25
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_25'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_26'/>
26
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_26'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.DNACompoundSet;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_27'/>
27
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_27'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.transcription.TranscriptionEngine;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_28'/>
28
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_28'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.slf4j.Logger;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_29'/>
29
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_29'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.slf4j.LoggerFactory;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_30'/>
30
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_30'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_31'/>
31
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_31'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.ArrayList;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_32'/>
32
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_32'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.Collections;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_33'/>
33
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_33'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.LinkedHashMap;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_34'/>
34
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_34'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_35'/>
35
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_35'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_36'/>
36
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_36'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * This is the sequence if you want to go from a gene sequence to a protein sequence. Need to start with a</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_37'/>
37
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_37'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * ChromosomeSequence then getting a GeneSequence and then a TranscriptSequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_38'/>
38
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_38'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * @author Scooter Willis</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_39'/>
39
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_39'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_40'/>
40
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_40'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>public class TranscriptSequence extends DNASequence {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_41'/>
41
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_41'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_42'/>
42
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_42'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;private final static Logger logger = LoggerFactory.getLogger(TranscriptSequence.class);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_43'/>
43
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_43'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_44'/>
44
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_44'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;private final ArrayList&#60;CDSSequence&#62; cdsSequenceList = new ArrayList&#60;CDSSequence&#62;();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_45'/>
45
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_45'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;private final LinkedHashMap&#60;String, CDSSequence&#62; cdsSequenceHashMap = new LinkedHashMap&#60;String, CDSSequence&#62;();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_46'/>
46
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_46'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;private StartCodonSequence startCodonSequence = null;</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_47'/>
47
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_47'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;private StopCodonSequence stopCodonSequence = null;</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_48'/>
48
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_48'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;private GeneSequence parentGeneSequence = null;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_49'/>
49
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_49'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_50'/>
50
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_50'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_51'/>
51
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_51'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_52'/>
52
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_52'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param parentDNASequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_53'/>
53
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_53'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param begin</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_54'/>
54
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_54'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param end inclusive of end</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_55'/>
55
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_55'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_56'/>
56
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_56'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;public TranscriptSequence(GeneSequence parentDNASequence, int begin, int end) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_57'/>
57
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_57'>1</a>
<span>
1. <init> : removed call to org/biojava/nbio/core/sequence/TranscriptSequence::setParentSequence &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;setParentSequence(parentDNASequence);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_58'/>
58
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_58'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;this.parentGeneSequence = parentDNASequence;</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_59'/>
59
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_59'>1</a>
<span>
1. <init> : removed call to org/biojava/nbio/core/sequence/TranscriptSequence::setBioBegin &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;setBioBegin(begin);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_60'/>
60
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_60'>1</a>
<span>
1. <init> : removed call to org/biojava/nbio/core/sequence/TranscriptSequence::setBioEnd &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;setBioEnd(end);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_61'/>
61
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_61'>1</a>
<span>
1. <init> : removed call to org/biojava/nbio/core/sequence/TranscriptSequence::setCompoundSet &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;this.setCompoundSet(DNACompoundSet.getDNACompoundSet());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_62'/>
62
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_62'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_63'/>
63
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_63'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_64'/>
64
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_64'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_65'/>
65
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_65'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;@Override</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_66'/>
66
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_66'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public int getLength() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_67'/>
67
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_67'>3</a>
<span>
1. getLength : Replaced integer subtraction with addition &rarr; NO_COVERAGE<br/>
2. getLength : Replaced integer addition with subtraction &rarr; NO_COVERAGE<br/>
3. getLength : replaced return of integer sized value with (x == 0 ? 1 : 0) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return Math.abs(this.getBioEnd() - this.getBioBegin()) + 1;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_68'/>
68
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_68'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_69'/>
69
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_69'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_70'/>
70
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_70'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_71'/>
71
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_71'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return the strand</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_72'/>
72
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_72'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_73'/>
73
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_73'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public Strand getStrand() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_74'/>
74
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_74'>1</a>
<span>
1. getStrand : mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getStrand to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return parentGeneSequence.getStrand();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_75'/>
75
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_75'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_76'/>
76
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_76'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_77'/>
77
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_77'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_78'/>
78
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_78'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Remove a CDS or coding sequence from the transcript sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_79'/>
79
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_79'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param accession</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_80'/>
80
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_80'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_81'/>
81
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_81'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_82'/>
82
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_82'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public CDSSequence removeCDS(String accession) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_83'/>
83
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_83'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;for (CDSSequence cdsSequence : cdsSequenceList) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_84'/>
84
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_84'>1</a>
<span>
1. removeCDS : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;if (cdsSequence.getAccession().getID().equals(accession)) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_85'/>
85
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_85'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;cdsSequenceList.remove(cdsSequence);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_86'/>
86
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_86'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;cdsSequenceHashMap.remove(accession);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_87'/>
87
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_87'>1</a>
<span>
1. removeCDS : mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::removeCDS to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;return cdsSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_88'/>
88
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_88'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_89'/>
89
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_89'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_90'/>
90
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_90'>1</a>
<span>
1. removeCDS : mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::removeCDS to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return null;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_91'/>
91
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_91'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_92'/>
92
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_92'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_93'/>
93
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_93'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_94'/>
94
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_94'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Get the CDS sequences that have been added to the TranscriptSequences</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_95'/>
95
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_95'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_96'/>
96
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_96'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_97'/>
97
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_97'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public LinkedHashMap&#60;String, CDSSequence&#62; getCDSSequences() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_98'/>
98
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_98'>1</a>
<span>
1. getCDSSequences : mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getCDSSequences to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return cdsSequenceHashMap;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_99'/>
99
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_99'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_100'/>
100
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_100'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_101'/>
101
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_101'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_102'/>
102
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_102'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Add a Coding Sequence region with phase to the transcript sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_103'/>
103
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_103'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param accession</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_104'/>
104
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_104'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param begin</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_105'/>
105
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_105'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param end</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_106'/>
106
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_106'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param phase 0,1,2</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_107'/>
107
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_107'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_108'/>
108
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_108'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_109'/>
109
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_109'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public CDSSequence addCDS(AccessionID accession, int begin, int end, int phase) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_110'/>
110
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_110'>1</a>
<span>
1. addCDS : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;if (cdsSequenceHashMap.containsKey(accession.getID())) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_111'/>
111
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_111'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;throw new Exception(&#34;Duplicate accesion id &#34; + accession.getID());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_112'/>
112
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_112'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_113'/>
113
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_113'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;CDSSequence cdsSequence = new CDSSequence(this, begin, end, phase); //sense should be the same as parent</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_114'/>
114
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_114'>1</a>
<span>
1. addCDS : removed call to org/biojava/nbio/core/sequence/CDSSequence::setAccession &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;cdsSequence.setAccession(accession);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_115'/>
115
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_115'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;cdsSequenceList.add(cdsSequence);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_116'/>
116
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_116'>1</a>
<span>
1. addCDS : removed call to java/util/Collections::sort &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;Collections.sort(cdsSequenceList, new CDSComparator());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_117'/>
117
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_117'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;cdsSequenceHashMap.put(accession.getID(), cdsSequence);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_118'/>
118
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_118'>1</a>
<span>
1. addCDS : mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::addCDS to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return cdsSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_119'/>
119
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_119'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_120'/>
120
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_120'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_121'/>
121
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_121'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_122'/>
122
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_122'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * http://www.sequenceontology.org/gff3.shtml</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_123'/>
123
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_123'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * http://biowiki.org/~yam/bioe131/GFF.ppt</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_124'/>
124
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_124'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_125'/>
125
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_125'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_126'/>
126
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_126'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_127'/>
127
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_127'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Return a list of protein sequences based on each CDS sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_128'/>
128
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_128'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * where the phase shift between two CDS sequences is assigned to the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_129'/>
129
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_129'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * CDS sequence that starts the triplet. This can be used to map</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_130'/>
130
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_130'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * a CDS/exon region of a protein sequence back to the DNA sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_131'/>
131
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_131'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * If you have a protein sequence and a predicted gene you can take the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_132'/>
132
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_132'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * predict CDS protein sequences and align back to the protein sequence.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_133'/>
133
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_133'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * If you have errors in mapping the predicted protein CDS regions to</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_134'/>
134
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_134'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * an the known protein sequence then you can identify possible errors</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_135'/>
135
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_135'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * in the prediction</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_136'/>
136
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_136'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_137'/>
137
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_137'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_138'/>
138
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_138'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_139'/>
139
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_139'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public ArrayList&#60;ProteinSequence&#62; getProteinCDSSequences() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_140'/>
140
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_140'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;ArrayList&#60;ProteinSequence&#62; proteinSequenceList = new ArrayList&#60;ProteinSequence&#62;();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_141'/>
141
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_141'>2</a>
<span>
1. getProteinCDSSequences : changed conditional boundary &rarr; NO_COVERAGE<br/>
2. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;for (int i = 0; i &#60; cdsSequenceList.size(); i++) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_142'/>
142
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_142'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;CDSSequence cdsSequence = cdsSequenceList.get(i);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_143'/>
143
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_143'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;String codingSequence = cdsSequence.getCodingSequence();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_144'/>
144
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_144'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;//          logger.debug(&#34;CDS {} {} = {}&#34;, getStrand(), cdsSequence.getPhase(), codingSequence);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_145'/>
145
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_145'>1</a>
<span>
1. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;if (this.getStrand() == Strand.NEGATIVE) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_146'/>
146
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_146'>1</a>
<span>
1. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;if (cdsSequence.phase == 1) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_147'/>
147
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_147'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;codingSequence = codingSequence.substring(1, codingSequence.length());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_148'/>
148
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_148'>1</a>
<span>
1. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;} else if (cdsSequence.phase == 2) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_149'/>
149
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_149'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;codingSequence = codingSequence.substring(2, codingSequence.length());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_150'/>
150
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_150'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_151'/>
151
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_151'>3</a>
<span>
1. getProteinCDSSequences : changed conditional boundary &rarr; NO_COVERAGE<br/>
2. getProteinCDSSequences : Replaced integer subtraction with addition &rarr; NO_COVERAGE<br/>
3. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;if (i &#60; cdsSequenceList.size() - 1) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_152'/>
152
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_152'>1</a>
<span>
1. getProteinCDSSequences : Replaced integer addition with subtraction &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;&#9;CDSSequence nextCDSSequence = cdsSequenceList.get(i + 1);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_153'/>
153
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_153'>1</a>
<span>
1. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;&#9;if (nextCDSSequence.phase == 1) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_154'/>
154
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_154'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;String nextCodingSequence = nextCDSSequence.getCodingSequence();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_155'/>
155
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_155'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;codingSequence = codingSequence + nextCodingSequence.substring(0, 1);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_156'/>
156
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_156'>1</a>
<span>
1. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;&#9;} else if (nextCDSSequence.phase == 2) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_157'/>
157
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_157'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;String nextCodingSequence = nextCDSSequence.getCodingSequence();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_158'/>
158
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_158'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;codingSequence = codingSequence + nextCodingSequence.substring(0, 2);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_159'/>
159
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_159'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_160'/>
160
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_160'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_161'/>
161
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_161'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;} else {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_162'/>
162
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_162'>1</a>
<span>
1. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;if (cdsSequence.phase == 1) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_163'/>
163
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_163'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;codingSequence = codingSequence.substring(1, codingSequence.length());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_164'/>
164
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_164'>1</a>
<span>
1. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;} else if (cdsSequence.phase == 2) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_165'/>
165
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_165'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;codingSequence = codingSequence.substring(2, codingSequence.length());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_166'/>
166
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_166'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_167'/>
167
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_167'>3</a>
<span>
1. getProteinCDSSequences : changed conditional boundary &rarr; NO_COVERAGE<br/>
2. getProteinCDSSequences : Replaced integer subtraction with addition &rarr; NO_COVERAGE<br/>
3. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;if (i &#60; cdsSequenceList.size() - 1) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_168'/>
168
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_168'>1</a>
<span>
1. getProteinCDSSequences : Replaced integer addition with subtraction &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;&#9;CDSSequence nextCDSSequence = cdsSequenceList.get(i + 1);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_169'/>
169
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_169'>1</a>
<span>
1. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;&#9;if (nextCDSSequence.phase == 1) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_170'/>
170
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_170'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;String nextCodingSequence = nextCDSSequence.getCodingSequence();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_171'/>
171
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_171'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;codingSequence = codingSequence + nextCodingSequence.substring(0, 1);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_172'/>
172
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_172'>1</a>
<span>
1. getProteinCDSSequences : negated conditional &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;&#9;} else if (nextCDSSequence.phase == 2) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_173'/>
173
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_173'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;String nextCodingSequence = nextCDSSequence.getCodingSequence();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_174'/>
174
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_174'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9;codingSequence = codingSequence + nextCodingSequence.substring(0, 2);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_175'/>
175
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_175'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_176'/>
176
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_176'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_177'/>
177
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_177'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_178'/>
178
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_178'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_179'/>
179
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_179'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_180'/>
180
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_180'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;//    logger.debug(&#34;Coding Sequence: {}&#34;, codingSequence);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_181'/>
181
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_181'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_182'/>
182
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_182'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;DNASequence dnaCodingSequence = null;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_183'/>
183
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_183'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;try {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_184'/>
184
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_184'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;dnaCodingSequence = new DNASequence(codingSequence.toUpperCase());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_185'/>
185
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_185'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;} catch (CompoundNotFoundException e) {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_186'/>
186
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_186'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;// if I understand this should not happen, please correct if I&#39;m wrong - JD 2014-10-24</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_187'/>
187
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_187'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;&#9;logger.error(&#34;Could not create DNA coding sequence, {}. This is most likely a bug.&#34;, e.getMessage());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_188'/>
188
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_188'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_189'/>
189
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_189'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;RNASequence rnaCodingSequence = dnaCodingSequence.getRNASequence(TranscriptionEngine.getDefault());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_190'/>
190
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_190'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;ProteinSequence proteinSequence = rnaCodingSequence.getProteinSequence(TranscriptionEngine.getDefault());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_191'/>
191
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_191'>1</a>
<span>
1. getProteinCDSSequences : removed call to org/biojava/nbio/core/sequence/ProteinSequence::setAccession &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;proteinSequence.setAccession(new AccessionID(cdsSequence.getAccession().getID()));</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_192'/>
192
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_192'>1</a>
<span>
1. getProteinCDSSequences : removed call to org/biojava/nbio/core/sequence/ProteinSequence::setParentDNASequence &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;&#9;proteinSequence.setParentDNASequence(cdsSequence, 1, cdsSequence.getLength());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_193'/>
193
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_193'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;proteinSequenceList.add(proteinSequence);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_194'/>
194
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_194'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_195'/>
195
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_195'>1</a>
<span>
1. getProteinCDSSequences : mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getProteinCDSSequences to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return proteinSequenceList;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_196'/>
196
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_196'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_197'/>
197
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_197'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_198'/>
198
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_198'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_199'/>
199
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_199'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Get the stitched together CDS sequences then maps to the cDNA</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_200'/>
200
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_200'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_201'/>
201
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_201'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_202'/>
202
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_202'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public DNASequence getDNACodingSequence() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_203'/>
203
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_203'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;StringBuilder sb = new StringBuilder();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_204'/>
204
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_204'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;for (CDSSequence cdsSequence : cdsSequenceList) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_205'/>
205
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_205'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;sb.append(cdsSequence.getCodingSequence());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_206'/>
206
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_206'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_207'/>
207
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_207'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_208'/>
208
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_208'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;DNASequence dnaSequence = null;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_209'/>
209
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_209'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;try {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_210'/>
210
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_210'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;dnaSequence = new DNASequence(sb.toString().toUpperCase());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_211'/>
211
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_211'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;} catch (CompoundNotFoundException e) {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_212'/>
212
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_212'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;// if I understand this should not happen, please correct if I&#39;m wrong - JD 2014-10-24</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_213'/>
213
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_213'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;logger.error(&#34;Could not create DNA coding sequence, {}. This is most likely a bug.&#34;, e.getMessage());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_214'/>
214
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_214'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_215'/>
215
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_215'>1</a>
<span>
1. getDNACodingSequence : removed call to org/biojava/nbio/core/sequence/DNASequence::setAccession &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;dnaSequence.setAccession(new AccessionID(this.getAccession().getID()));</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_216'/>
216
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_216'>1</a>
<span>
1. getDNACodingSequence : mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getDNACodingSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return dnaSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_217'/>
217
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_217'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_218'/>
218
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_218'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_219'/>
219
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_219'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_220'/>
220
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_220'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Get the protein sequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_221'/>
221
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_221'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_222'/>
222
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_222'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_223'/>
223
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_223'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public ProteinSequence getProteinSequence() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_224'/>
224
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_224'>1</a>
<span>
1. getProteinSequence : mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getProteinSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return getProteinSequence(TranscriptionEngine.getDefault());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_225'/>
225
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_225'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_226'/>
226
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_226'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_227'/>
227
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_227'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_228'/>
228
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_228'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Get the protein sequence with user defined TranscriptEngine</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_229'/>
229
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_229'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param engine</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_230'/>
230
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_230'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_231'/>
231
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_231'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_232'/>
232
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_232'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public ProteinSequence getProteinSequence(TranscriptionEngine engine) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_233'/>
233
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_233'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;DNASequence dnaCodingSequence = getDNACodingSequence();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_234'/>
234
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_234'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;RNASequence rnaCodingSequence = dnaCodingSequence.getRNASequence(engine);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_235'/>
235
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_235'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;ProteinSequence proteinSequence = rnaCodingSequence.getProteinSequence(engine);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_236'/>
236
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_236'>1</a>
<span>
1. getProteinSequence : removed call to org/biojava/nbio/core/sequence/ProteinSequence::setAccession &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;proteinSequence.setAccession(new AccessionID(this.getAccession().getID()));</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_237'/>
237
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_237'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_238'/>
238
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_238'>1</a>
<span>
1. getProteinSequence : mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getProteinSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return proteinSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_239'/>
239
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_239'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_240'/>
240
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_240'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_241'/>
241
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_241'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_242'/>
242
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_242'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return the startCodonSequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_243'/>
243
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_243'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_244'/>
244
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_244'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public StartCodonSequence getStartCodonSequence() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_245'/>
245
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_245'>1</a>
<span>
1. getStartCodonSequence : mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getStartCodonSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return startCodonSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_246'/>
246
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_246'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_247'/>
247
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_247'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_248'/>
248
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_248'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_249'/>
249
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_249'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param startCodonSequence the startCodonSequence to set</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_250'/>
250
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_250'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_251'/>
251
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_251'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public void addStartCodonSequence(AccessionID accession, int begin, int end) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_252'/>
252
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_252'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;this.startCodonSequence = new StartCodonSequence(this, begin, end);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_253'/>
253
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_253'>1</a>
<span>
1. addStartCodonSequence : removed call to org/biojava/nbio/core/sequence/StartCodonSequence::setAccession &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;startCodonSequence.setAccession(accession);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_254'/>
254
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_254'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_255'/>
255
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_255'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_256'/>
256
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_256'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_257'/>
257
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_257'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return the stopCodonSequence</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_258'/>
258
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_258'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_259'/>
259
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_259'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public StopCodonSequence getStopCodonSequence() {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_260'/>
260
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_260'>1</a>
<span>
1. getStopCodonSequence : mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getStopCodonSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;return stopCodonSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_261'/>
261
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_261'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_262'/>
262
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_262'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_263'/>
263
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_263'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_264'/>
264
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_264'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param stopCodonSequence the stopCodonSequence to set</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_265'/>
265
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_265'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_266'/>
266
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_266'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public void addStopCodonSequence(AccessionID accession, int begin, int end) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_267'/>
267
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_267'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;this.stopCodonSequence = new StopCodonSequence(this, begin, end);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_268'/>
268
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_268'>1</a>
<span>
1. addStopCodonSequence : removed call to org/biojava/nbio/core/sequence/StopCodonSequence::setAccession &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;stopCodonSequence.setAccession(accession);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_269'/>
269
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_269'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_270'/>
270
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_270'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>}</span></pre></td></tr>



<tr><td></td><td></td><td><h2>Mutations</h2></td></tr>

<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_57'>57</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_57'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b><init><br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/TranscriptSequence::setParentSequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_59'>59</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_59'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b><init><br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/TranscriptSequence::setBioBegin &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_60'>60</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_60'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b><init><br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/TranscriptSequence::setBioEnd &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_61'>61</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_61'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b><init><br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/TranscriptSequence::setCompoundSet &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_67'>67</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_67'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getLength<br/><b>Killed by : </b>none</span></span> Replaced integer subtraction with addition &rarr; NO_COVERAGE</p> <p class='NO_COVERAGE'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>getLength<br/><b>Killed by : </b>none</span></span> Replaced integer addition with subtraction &rarr; NO_COVERAGE</p> <p class='NO_COVERAGE'><span class='pop'>3.<span><b>3</b><br/><b>Location : </b>getLength<br/><b>Killed by : </b>none</span></span> replaced return of integer sized value with (x == 0 ? 1 : 0) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_74'>74</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_74'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getStrand<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getStrand to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_84'>84</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_84'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>removeCDS<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_87'>87</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_87'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>removeCDS<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::removeCDS to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_90'>90</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_90'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>removeCDS<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::removeCDS to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_98'>98</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_98'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getCDSSequences<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getCDSSequences to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_110'>110</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_110'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>addCDS<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_114'>114</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_114'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>addCDS<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/CDSSequence::setAccession &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_116'>116</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_116'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>addCDS<br/><b>Killed by : </b>none</span></span> removed call to java/util/Collections::sort &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_118'>118</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_118'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>addCDS<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::addCDS to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_141'>141</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_141'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> changed conditional boundary &rarr; NO_COVERAGE</p> <p class='NO_COVERAGE'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_145'>145</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_145'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_146'>146</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_146'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_148'>148</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_148'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_151'>151</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_151'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> changed conditional boundary &rarr; NO_COVERAGE</p> <p class='NO_COVERAGE'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> Replaced integer subtraction with addition &rarr; NO_COVERAGE</p> <p class='NO_COVERAGE'><span class='pop'>3.<span><b>3</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_152'>152</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_152'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> Replaced integer addition with subtraction &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_153'>153</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_153'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_156'>156</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_156'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_162'>162</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_162'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_164'>164</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_164'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_167'>167</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_167'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> changed conditional boundary &rarr; NO_COVERAGE</p> <p class='NO_COVERAGE'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> Replaced integer subtraction with addition &rarr; NO_COVERAGE</p> <p class='NO_COVERAGE'><span class='pop'>3.<span><b>3</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_168'>168</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_168'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> Replaced integer addition with subtraction &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_169'>169</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_169'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_172'>172</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_172'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> negated conditional &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_191'>191</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_191'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/ProteinSequence::setAccession &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_192'>192</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_192'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/ProteinSequence::setParentDNASequence &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_195'>195</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_195'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinCDSSequences<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getProteinCDSSequences to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_215'>215</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_215'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getDNACodingSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/DNASequence::setAccession &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_216'>216</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_216'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getDNACodingSequence<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getDNACodingSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_224'>224</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_224'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinSequence<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getProteinSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_236'>236</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_236'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/ProteinSequence::setAccession &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_238'>238</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_238'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getProteinSequence<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getProteinSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_245'>245</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_245'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getStartCodonSequence<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getStartCodonSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_253'>253</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_253'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>addStartCodonSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/StartCodonSequence::setAccession &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_260'>260</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_260'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>getStopCodonSequence<br/><b>Killed by : </b>none</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/TranscriptSequence::getStopCodonSequence to ( if (x != null) null else throw new RuntimeException ) &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@2bec8fc3_268'>268</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@2bec8fc3_268'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>addStopCodonSequence<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/StopCodonSequence::setAccession &rarr; NO_COVERAGE</p> 
</td>
</tr>

</table>


<h2>Active mutators</h2>
<ul>
<li class='mutator'>INCREMENTS_MUTATOR</li>
<li class='mutator'>VOID_METHOD_CALL_MUTATOR</li>
<li class='mutator'>RETURN_VALS_MUTATOR</li>
<li class='mutator'>MATH_MUTATOR</li>
<li class='mutator'>NEGATE_CONDITIONALS_MUTATOR</li>
<li class='mutator'>INVERT_NEGS_MUTATOR</li>
<li class='mutator'>CONDITIONALS_BOUNDARY_MUTATOR</li>

</ul>

<h2>Tests examined</h2>
<ul>
</ul>

<br/>

Report generated by <a href='http://pitest.org'>PIT</a> 1.4.0

</body>
</html>