<!DOCTYPE html>
<html>
<head>
    <link rel="stylesheet" type="text/css" href="../style.css">
</head>
<body>


<h1>GenbankReader.java</h1>

<table class="src">



<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_1'/>
1
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_1'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>/*</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_2'/>
2
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_2'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *                    BioJava development code</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_3'/>
3
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_3'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_4'/>
4
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_4'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * This code may be freely distributed and modified under the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_5'/>
5
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_5'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * terms of the GNU Lesser General Public Licence.  This should</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_6'/>
6
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_6'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * be distributed with the code.  If you do not have a copy,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_7'/>
7
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_7'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * see:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_8'/>
8
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_8'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_9'/>
9
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_9'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.gnu.org/copyleft/lesser.html</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_10'/>
10
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_10'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_11'/>
11
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_11'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * Copyright for this code is held jointly by the individual</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_12'/>
12
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_12'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * authors.  These should be listed in @author doc comments.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_13'/>
13
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_13'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_14'/>
14
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_14'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * @author Scooter Willis ;lt;willishf at gmail dot com&#38;gt;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_15'/>
15
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_15'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * @author Karl Nicholas &#60;github:karlnicholas&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_16'/>
16
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_16'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * @author Paolo Pavan</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_17'/>
17
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_17'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_18'/>
18
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_18'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * For more information on the BioJava project and its aims,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_19'/>
19
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_19'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * or to join the biojava-l mailing list, visit the home page</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_20'/>
20
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_20'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * at:</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_21'/>
21
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_21'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_22'/>
22
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_22'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *      http://www.biojava.org/</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_23'/>
23
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_23'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_24'/>
24
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_24'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * Created on 01-21-2010</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_25'/>
25
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_25'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_26'/>
26
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_26'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>package org.biojava.nbio.core.sequence.io;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_27'/>
27
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_27'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_28'/>
28
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_28'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_29'/>
29
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_29'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.DNASequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_30'/>
30
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_30'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.DataSource;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_31'/>
31
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_31'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.ProteinSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_32'/>
32
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_32'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.TaxonomyID;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_33'/>
33
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_33'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_34'/>
34
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_34'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_35'/>
35
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_35'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.DNACompoundSet;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_36'/>
36
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_36'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_37'/>
37
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_37'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.features.AbstractFeature;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_38'/>
38
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_38'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.features.DBReferenceInfo;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_39'/>
39
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_39'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_40'/>
40
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_40'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_41'/>
41
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_41'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.template.AbstractSequence;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_42'/>
42
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_42'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.biojava.nbio.core.sequence.template.Compound;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_43'/>
43
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_43'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.slf4j.Logger;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_44'/>
44
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_44'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import org.slf4j.LoggerFactory;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_45'/>
45
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_45'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_46'/>
46
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_46'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import java.io.*;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_47'/>
47
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_47'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.ArrayList;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_48'/>
48
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_48'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.HashMap;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_49'/>
49
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_49'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>import java.util.LinkedHashMap;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_50'/>
50
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_50'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_51'/>
51
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_51'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_52'/>
52
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_52'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * Use GenbankReaderHelper as an example of how to use this class where GenbankReaderHelper should be the</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_53'/>
53
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_53'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> * primary class used to read Genbank files</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_54'/>
54
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_54'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_55'/>
55
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_55'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''> */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_56'/>
56
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_56'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>public class GenbankReader&#60;S extends AbstractSequence&#60;C&#62;, C extends Compound&#62; {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_57'/>
57
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_57'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_58'/>
58
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_58'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;private SequenceCreatorInterface&#60;C&#62; sequenceCreator;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_59'/>
59
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_59'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;private GenbankSequenceParser&#60;S,C&#62; genbankParser;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_60'/>
60
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_60'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;private BufferedReader bufferedReader;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_61'/>
61
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_61'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;private boolean closed;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_62'/>
62
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_62'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;private final Logger logger = LoggerFactory.getLogger(this.getClass());</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_63'/>
63
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_63'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_64'/>
64
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_64'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public boolean isClosed() {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_65'/>
65
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_65'>1</a>
<span>
1. isClosed : replaced return of integer sized value with (x == 0 ? 1 : 0) &rarr; KILLED<br/>

</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return closed;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_66'/>
66
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_66'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_67'/>
67
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_67'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_68'/>
68
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_68'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_69'/>
69
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_69'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because we need details about</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_70'/>
70
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_70'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * local file offsets for quick reads. InputStreams does not give you the name of the stream to access quickly via file seek. A seek in</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_71'/>
71
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_71'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * an inputstream is forced to read all the data so you don&#39;t gain anything.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_72'/>
72
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_72'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param is</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_73'/>
73
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_73'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param headerParser</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_74'/>
74
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_74'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequenceCreator</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_75'/>
75
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_75'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_76'/>
76
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_76'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public GenbankReader(final InputStream is, final SequenceHeaderParserInterface&#60;S,C&#62; headerParser,</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_77'/>
77
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_77'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;&#9; final SequenceCreatorInterface&#60;C&#62; sequenceCreator) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_78'/>
78
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_78'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;this.sequenceCreator = sequenceCreator;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_79'/>
79
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_79'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;bufferedReader = new BufferedReader(new InputStreamReader(is));</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_80'/>
80
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_80'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;genbankParser = new GenbankSequenceParser&#60;&#62;();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_81'/>
81
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_81'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;closed = false;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_82'/>
82
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_82'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_83'/>
83
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_83'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_84'/>
84
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_84'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_85'/>
85
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_85'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * If you are going to use the FileProxyProteinSequenceCreator then you</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_86'/>
86
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_86'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * need to use this constructor because we need details about</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_87'/>
87
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_87'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * the location of the file.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_88'/>
88
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_88'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param file</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_89'/>
89
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_89'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param headerParser</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_90'/>
90
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_90'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param sequenceCreator</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_91'/>
91
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_91'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws FileNotFoundException if the file does not exist, is a directory</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_92'/>
92
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_92'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * &#9;rather than a regular file, or for some other reason cannot be opened</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_93'/>
93
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_93'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * &#9;for reading.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_94'/>
94
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_94'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws SecurityException if a security manager exists and its checkRead</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_95'/>
95
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_95'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * &#9;method denies read access to the file.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_96'/>
96
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_96'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_97'/>
97
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_97'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public GenbankReader(</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_98'/>
98
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_98'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;final File file,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_99'/>
99
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_99'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;final SequenceHeaderParserInterface&#60;S,C&#62; headerParser,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_100'/>
100
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_100'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;final SequenceCreatorInterface&#60;C&#62; sequenceCreator</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_101'/>
101
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_101'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;) throws FileNotFoundException {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_102'/>
102
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_102'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_103'/>
103
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_103'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;this.bufferedReader = new BufferedReader(new FileReader(file));</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_104'/>
104
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_104'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;this.sequenceCreator = sequenceCreator;</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_105'/>
105
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_105'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;genbankParser = new GenbankSequenceParser&#60;&#62;();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_106'/>
106
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_106'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_107'/>
107
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_107'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_108'/>
108
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_108'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_109'/>
109
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_109'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * The parsing is done in this method.&#60;br&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_110'/>
110
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_110'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * This method tries to process all the available Genbank records</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_111'/>
111
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_111'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * in the File or InputStream, closes the underlying resource,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_112'/>
112
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_112'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * and return the results in {@link LinkedHashMap}.&#60;br&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_113'/>
113
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_113'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * You don&#39;t need to call {@link #close()} after calling this method.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_114'/>
114
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_114'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @see #process(int)</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_115'/>
115
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_115'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return {@link HashMap} containing all the parsed Genbank records</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_116'/>
116
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_116'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * present, starting current fileIndex onwards.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_117'/>
117
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_117'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws IOException</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_118'/>
118
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_118'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws CompoundNotFoundException</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_119'/>
119
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_119'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_120'/>
120
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_120'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public LinkedHashMap&#60;String,S&#62; process() throws IOException, CompoundNotFoundException {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_121'/>
121
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_121'>1</a>
<span>
1. process : mutated return of Object value for org/biojava/nbio/core/sequence/io/GenbankReader::process to ( if (x != null) null else throw new RuntimeException ) &rarr; KILLED<br/>

</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return process(-1);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_122'/>
122
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_122'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_123'/>
123
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_123'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_124'/>
124
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_124'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;/**</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_125'/>
125
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_125'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * This method tries to parse maximum &#60;code&#62;max&#60;/code&#62; records from</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_126'/>
126
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_126'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * the open File or InputStream, and leaves the underlying resource open.&#60;br&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_127'/>
127
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_127'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; *</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_128'/>
128
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_128'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * Subsequent calls to the same method continue parsing the rest of the file.&#60;br&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_129'/>
129
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_129'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * This is particularly useful when dealing with very big data files,</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_130'/>
130
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_130'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * (e.g. NCBI nr database), which can&#39;t fit into memory and will take long</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_131'/>
131
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_131'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * time before the first result is available.&#60;br&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_132'/>
132
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_132'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * &#60;b&#62;N.B.&#60;/b&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_133'/>
133
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_133'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * &#60;ul&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_134'/>
134
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_134'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * &#60;li&#62;This method can&#39;t be called after calling its NO-ARGUMENT twin.&#60;/li&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_135'/>
135
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_135'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * &#60;li&#62;remember to close the underlying resource when you are done.&#60;/li&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_136'/>
136
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_136'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * &#60;/ul&#62;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_137'/>
137
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_137'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @see #process()</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_138'/>
138
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_138'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @author Amr AL-Hossary</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_139'/>
139
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_139'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @since 3.0.6</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_140'/>
140
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_140'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @param max maximum number of records to return, &#60;code&#62;-1&#60;/code&#62; for infinity.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_141'/>
141
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_141'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @return {@link HashMap} containing maximum &#60;code&#62;max&#60;/code&#62; parsed Genbank records</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_142'/>
142
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_142'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * present, starting current fileIndex onwards.</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_143'/>
143
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_143'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws IOException</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_144'/>
144
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_144'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; * @throws CompoundNotFoundException</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_145'/>
145
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_145'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9; */</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_146'/>
146
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_146'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public LinkedHashMap&#60;String,S&#62; process(final int max) throws IOException, CompoundNotFoundException {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_147'/>
147
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_147'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;LinkedHashMap&#60;String,S&#62; sequences = new LinkedHashMap&#60;&#62;();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_148'/>
148
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_148'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;@SuppressWarnings(&#34;unchecked&#34;)</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_149'/>
149
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_149'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;int i=0;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_150'/>
150
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_150'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;while(true) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_151'/>
151
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_151'>4</a>
<span>
1. process : changed conditional boundary &rarr; SURVIVED<br/>
2. process : changed conditional boundary &rarr; KILLED<br/>
3. process : negated conditional &rarr; KILLED<br/>
4. process : negated conditional &rarr; KILLED<br/>

</span>
</td>
<td class='covered'><pre><span  class='survived'>&#9;&#9;&#9;if(max&#62;0 &#38;&#38; i&#62;=max) break;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_152'/>
152
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_152'>1</a>
<span>
1. process : Changed increment from 1 to -1 &rarr; KILLED<br/>

</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;i++;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_153'/>
153
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_153'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;String seqString = genbankParser.getSequence(bufferedReader, 0);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_154'/>
154
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_154'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;//reached end of file?</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_155'/>
155
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_155'>1</a>
<span>
1. process : negated conditional &rarr; KILLED<br/>

</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;if(seqString==null) break;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_156'/>
156
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_156'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>            @SuppressWarnings(&#34;unchecked&#34;)</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_157'/>
157
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_157'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;S sequence = (S) sequenceCreator.getSequence(seqString, 0);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_158'/>
158
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_158'>1</a>
<span>
1. process : removed call to org/biojava/nbio/core/sequence/io/GenericGenbankHeaderParser::parseHeader &rarr; KILLED<br/>

</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;genbankParser.getSequenceHeaderParser().parseHeader(genbankParser.getHeader(), sequence);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_159'/>
159
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_159'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_160'/>
160
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_160'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;// add features to new sequence</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_161'/>
161
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_161'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;for (String k: genbankParser.getFeatures().keySet()){</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_162'/>
162
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_162'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;for (AbstractFeature f: genbankParser.getFeatures(k)){</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_163'/>
163
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_163'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;&#9;&#9;//f.getLocations().setSequence(sequence);  // can&#39;t set proper sequence source to features. It is actually needed? Don&#39;t think so...</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_164'/>
164
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_164'>1</a>
<span>
1. process : removed call to org/biojava/nbio/core/sequence/template/AbstractSequence::addFeature &rarr; KILLED<br/>

</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;&#9;&#9;sequence.addFeature(f);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_165'/>
165
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_165'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_166'/>
166
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_166'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_167'/>
167
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_167'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_168'/>
168
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_168'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;// add taxonomy ID to new sequence</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_169'/>
169
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_169'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;ArrayList&#60;DBReferenceInfo&#62; dbQualifier = genbankParser.getDatabaseReferences().get(&#34;db_xref&#34;);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_170'/>
170
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_170'>1</a>
<span>
1. process : negated conditional &rarr; KILLED<br/>

</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;if (dbQualifier != null){</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_171'/>
171
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_171'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;&#9;DBReferenceInfo q = dbQualifier.get(0);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_172'/>
172
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_172'>1</a>
<span>
1. process : removed call to org/biojava/nbio/core/sequence/template/AbstractSequence::setTaxonomy &rarr; SURVIVED<br/>

</span>
</td>
<td class='covered'><pre><span  class='survived'>&#9;&#9;&#9;&#9;sequence.setTaxonomy(new TaxonomyID(q.getDatabase()+&#34;:&#34;+q.getId(), DataSource.GENBANK));</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_173'/>
173
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_173'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_174'/>
174
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_174'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_175'/>
175
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_175'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;sequences.put(sequence.getAccession().getID(), sequence);</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_176'/>
176
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_176'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_177'/>
177
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_177'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_178'/>
178
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_178'>2</a>
<span>
1. process : changed conditional boundary &rarr; SURVIVED<br/>
2. process : negated conditional &rarr; KILLED<br/>

</span>
</td>
<td class='covered'><pre><span  class='survived'>&#9;&#9;if (max &#60; 0) {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_179'/>
179
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_179'>1</a>
<span>
1. process : removed call to org/biojava/nbio/core/sequence/io/GenbankReader::close &rarr; KILLED<br/>

</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;&#9;close();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_180'/>
180
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_180'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_181'/>
181
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_181'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_182'/>
182
</td>
<td class='killed'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_182'>1</a>
<span>
1. process : mutated return of Object value for org/biojava/nbio/core/sequence/io/GenbankReader::process to ( if (x != null) null else throw new RuntimeException ) &rarr; KILLED<br/>

</span>
</td>
<td class='covered'><pre><span  class='killed'>&#9;&#9;return sequences;</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_183'/>
183
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_183'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_184'/>
184
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_184'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_185'/>
185
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_185'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public void close() {</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_186'/>
186
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_186'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;&#9;try {</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_187'/>
187
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_187'>1</a>
<span>
1. close : removed call to java/io/BufferedReader::close &rarr; SURVIVED<br/>

</span>
</td>
<td class='covered'><pre><span  class='survived'>&#9;&#9;&#9;bufferedReader.close();</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_188'/>
188
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_188'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;&#9;this.closed = true;</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_189'/>
189
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_189'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;} catch (IOException e) {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_190'/>
190
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_190'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;logger.error(&#34;Couldn&#39;t close the reader. {}&#34;, e.getMessage());</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_191'/>
191
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_191'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;&#9;this.closed = false;</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_192'/>
192
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_192'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;&#9;}</span></pre></td></tr>


<tr>
<td class='covered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_193'/>
193
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_193'></a>
<span>
</span>
</td>
<td class='covered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_194'/>
194
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_194'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_195'/>
195
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_195'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>&#9;public static void main(String[] args) throws Exception {</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_196'/>
196
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_196'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;String proteinFile = &#34;src/test/resources/BondFeature.gb&#34;;</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_197'/>
197
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_197'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;FileInputStream is = new FileInputStream(proteinFile);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_198'/>
198
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_198'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_199'/>
199
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_199'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;GenbankReader&#60;ProteinSequence, AminoAcidCompound&#62; proteinReader = new GenbankReader&#60;&#62;(is, new GenericGenbankHeaderParser&#60;&#62;(), new ProteinSequenceCreator(AminoAcidCompoundSet.getAminoAcidCompoundSet()));</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_200'/>
200
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_200'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;LinkedHashMap&#60;String,ProteinSequence&#62; proteinSequences = proteinReader.process();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_201'/>
201
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_201'>1</a>
<span>
1. main : removed call to java/io/PrintStream::println &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;System.out.println(proteinSequences);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_202'/>
202
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_202'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_203'/>
203
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_203'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;String inputFile = &#34;src/test/resources/NM_000266.gb&#34;;</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_204'/>
204
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_204'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;is = new FileInputStream(inputFile);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_205'/>
205
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_205'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;GenbankReader&#60;DNASequence, NucleotideCompound&#62; dnaReader = new GenbankReader&#60;&#62;(is, new GenericGenbankHeaderParser&#60;&#62;(), new DNASequenceCreator(DNACompoundSet.getDNACompoundSet()));</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_206'/>
206
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_206'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;LinkedHashMap&#60;String,DNASequence&#62; dnaSequences = dnaReader.process();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_207'/>
207
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_207'>1</a>
<span>
1. main : removed call to java/io/PrintStream::println &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;System.out.println(dnaSequences);</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_208'/>
208
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_208'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_209'/>
209
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_209'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;String crazyFile = &#34;src/test/resources/CraftedFeature.gb&#34;;</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_210'/>
210
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_210'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;is = new FileInputStream(crazyFile);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_211'/>
211
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_211'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;GenbankReader&#60;DNASequence, NucleotideCompound&#62; crazyReader = new GenbankReader&#60;&#62;(is, new GenericGenbankHeaderParser&#60;&#62;(), new DNASequenceCreator(DNACompoundSet.getDNACompoundSet()));</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_212'/>
212
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_212'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;&#9;LinkedHashMap&#60;String,DNASequence&#62; crazyAnnotatedSequences = crazyReader.process();</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_213'/>
213
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_213'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_214'/>
214
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_214'>1</a>
<span>
1. main : removed call to java/io/FileInputStream::close &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;is.close();</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_215'/>
215
</td>
<td class='survived'>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_215'>1</a>
<span>
1. main : removed call to java/io/PrintStream::println &rarr; NO_COVERAGE<br/>

</span>
</td>
<td class='uncovered'><pre><span  class='survived'>&#9;&#9;System.out.println(crazyAnnotatedSequences);</span></pre></td></tr>


<tr>
<td class='uncovered'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_216'/>
216
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_216'></a>
<span>
</span>
</td>
<td class='uncovered'><pre><span  class=''>&#9;}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_217'/>
217
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_217'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_218'/>
218
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_218'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''>}</span></pre></td></tr>


<tr>
<td class='na'>
<a name='org.pitest.mutationtest.report.html.SourceFile@265db36e_219'/>
219
</td>
<td class=''>
<span class='pop'>
<a href='#grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_219'></a>
<span>
</span>
</td>
<td class=''><pre><span  class=''></span></pre></td></tr>



<tr><td></td><td></td><td><h2>Mutations</h2></td></tr>

<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_65'>65</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_65'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>isClosed<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> replaced return of integer sized value with (x == 0 ? 1 : 0) &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_121'>121</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_121'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/io/GenbankReader::process to ( if (x != null) null else throw new RuntimeException ) &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_151'>151</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_151'/> 

<p class='SURVIVED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>process<br/><b>Killed by : </b>none</span></span> changed conditional boundary &rarr; SURVIVED</p> <p class='KILLED'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> changed conditional boundary &rarr; KILLED</p> <p class='KILLED'><span class='pop'>3.<span><b>3</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> negated conditional &rarr; KILLED</p> <p class='KILLED'><span class='pop'>4.<span><b>4</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_152'>152</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_152'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> Changed increment from 1 to -1 &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_155'>155</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_155'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_158'>158</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_158'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> removed call to org/biojava/nbio/core/sequence/io/GenericGenbankHeaderParser::parseHeader &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_164'>164</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_164'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> removed call to org/biojava/nbio/core/sequence/template/AbstractSequence::addFeature &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_170'>170</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_170'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_172'>172</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_172'/> 

<p class='SURVIVED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>process<br/><b>Killed by : </b>none</span></span> removed call to org/biojava/nbio/core/sequence/template/AbstractSequence::setTaxonomy &rarr; SURVIVED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_178'>178</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_178'/> 

<p class='SURVIVED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>process<br/><b>Killed by : </b>none</span></span> changed conditional boundary &rarr; SURVIVED</p> <p class='KILLED'><span class='pop'>2.<span><b>2</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> negated conditional &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_179'>179</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_179'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> removed call to org/biojava/nbio/core/sequence/io/GenbankReader::close &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_182'>182</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_182'/> 

<p class='KILLED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>process<br/><b>Killed by : </b>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest</span></span> mutated return of Object value for org/biojava/nbio/core/sequence/io/GenbankReader::process to ( if (x != null) null else throw new RuntimeException ) &rarr; KILLED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_187'>187</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_187'/> 

<p class='SURVIVED'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>close<br/><b>Killed by : </b>none</span></span> removed call to java/io/BufferedReader::close &rarr; SURVIVED</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_201'>201</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_201'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>main<br/><b>Killed by : </b>none</span></span> removed call to java/io/PrintStream::println &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_207'>207</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_207'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>main<br/><b>Killed by : </b>none</span></span> removed call to java/io/PrintStream::println &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_214'>214</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_214'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>main<br/><b>Killed by : </b>none</span></span> removed call to java/io/FileInputStream::close &rarr; NO_COVERAGE</p> 
</td>
</tr>
<tr>
<td><a href='#org.pitest.mutationtest.report.html.SourceFile@265db36e_215'>215</a></td> 
<td></td>
<td>

<a name='grouporg.pitest.mutationtest.report.html.SourceFile@265db36e_215'/> 

<p class='NO_COVERAGE'><span class='pop'>1.<span><b>1</b><br/><b>Location : </b>main<br/><b>Killed by : </b>none</span></span> removed call to java/io/PrintStream::println &rarr; NO_COVERAGE</p> 
</td>
</tr>

</table>


<h2>Active mutators</h2>
<ul>
<li class='mutator'>INCREMENTS_MUTATOR</li>
<li class='mutator'>VOID_METHOD_CALL_MUTATOR</li>
<li class='mutator'>RETURN_VALS_MUTATOR</li>
<li class='mutator'>MATH_MUTATOR</li>
<li class='mutator'>NEGATE_CONDITIONALS_MUTATOR</li>
<li class='mutator'>INVERT_NEGS_MUTATOR</li>
<li class='mutator'>CONDITIONALS_BOUNDARY_MUTATOR</li>

</ul>

<h2>Tests examined</h2>
<ul>
<li>org.biojava.nbio.core.sequence.io.GenbankWriterTest.testProcess(org.biojava.nbio.core.sequence.io.GenbankWriterTest) (14 ms)</li><li>org.biojava.nbio.core.sequence.io.GenbankReaderTest.org.biojava.nbio.core.sequence.io.GenbankReaderTest (27 ms)</li><li>org.biojava.nbio.core.sequence.io.GenbankCookbookTest.org.biojava.nbio.core.sequence.io.GenbankCookbookTest (1027 ms)</li>
</ul>

<br/>

Report generated by <a href='http://pitest.org'>PIT</a> 1.4.0

</body>
</html>